proto/xmtpv4/message_api/misbehavior_api.proto

```diff
@@ -0,0 +1,83 @@
+// API for reporting and querying node misbehavior in decentralized XMTP
+syntax = "proto3";
+
+package xmtp.xmtpv4.message_api;
+
+import "google/api/annotations.proto";
+import "identity/associations/signature.proto";
+import "xmtpv4/envelopes/envelopes.proto";
+import "xmtpv4/message_api/message_api.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/xmtpv4/message_api";
+
+enum Misbehavior {
+  MISBEHAVIOR_UNSPECIFIED = 0;
+  MISBEHAVIOR_UNRESPONSIVE_NODE = 1;
+  MISBEHAVIOR_SLOW_NODE = 2;
+  MISBEHAVIOR_FAILED_REQUEST = 3;
+  MISBEHAVIOR_OUT_OF_ORDER = 4;
+  MISBEHAVIOR_DUPLICATE_SEQUENCE_ID = 5;
+  MISBEHAVIOR_CAUSAL_ORDERING = 6;
+  MISBEHAVIOR_INVALID_PAYLOAD = 7;
+  MISBEHAVIOR_BLOCKCHAIN_INCONSISTENCY = 8;
+}
+
+message LivenessFailure {
+  uint32 response_time_ns = 1;
+  xmtp.xmtpv4.message_api.SubscribeEnvelopesRequest subscribe = 2;
+  xmtp.xmtpv4.message_api.QueryEnvelopesRequest query = 3;
+  xmtp.xmtpv4.message_api.PublishPayerEnvelopesRequest publish = 4;
+}
+
+message SafetyFailure {
+  repeated xmtp.xmtpv4.envelopes.OriginatorEnvelope envelopes = 1;
+}
+
+message UnsignedMisbehaviorReport {
+  uint64 reporter_time_ns = 1;
+  uint32 misbehaving_node_id = 2;
+  Misbehavior type = 3;
+  LivenessFailure liveness = 4;
```

Comment:
We don't want LivenessFailure and SafetyFailure to be a `oneof`?

Reply:
In most languages, `oneof` becomes a nested field. I suppose it's a question of developer ergonomics vs documenting expectations clearly, seeing as fields are optional either way. I'll go ahead and make it a `oneof`